src/constants.rs

```diff
@@ -3,4 +3,6 @@ use std::num::NonZeroUsize;
 // SAFETY: Safe because value non zero
 pub(crate) const MAX_BITS: NonZeroUsize = unsafe { NonZeroUsize::new_unchecked(255) };
 // SAFETY: Safe because value non zero
+pub(crate) const NUM_HASH_BITS: NonZeroUsize = unsafe { NonZeroUsize::new_unchecked(256) };
```

Comment:
use bit number less than the field size. if it is over size, it might wrap up to a smaller number.

Reply:
e.g. 253 bits

Reply:
Got it, the absolute MAX is larger than the modulus, so fewer bits are used to avoid going over the modulus.

WIP

Reply:
If we are going to support multiple curves, we will choose a smaller one than all of them, that might be the reason to choose 250 bits. Unless we dynamically calculate the bits based on the curve.